src/config.js

```diff
@@ -33,27 +31,61 @@ export function applyConfigToArgv({
         `specified in camel case: "${camelCase(option)}"`);
     }
 
-    if (option === commandExecuted) {
+    if (typeof options[option] === 'object' &&
+      typeof configObject[option] === 'object') {
+      // Descend into the nested configuration for a sub-command.
       newArgv = applyConfigToArgv({
         argv,
-        configObject: configObject[commandExecuted],
-        defaultValues: defaultValues[commandExecuted],
+        configObject: configObject[option],
+        options: options[option],
         configFileName});
       continue;
     }
 
+    const decamelizedOptName = decamelize(option, '-');
+    let expectedType;
+    if (options[decamelizedOptName]) {
```

Comment:
Some of these if statements are duplications of other ones. I think this is why you couldn't simplify them before. Here is a way to re-organize it a bit to allow for simplification:

```diff
diff --git a/src/config.js b/src/config.js
index f1f194b..e5c1598 100644
--- a/src/config.js
+++ b/src/config.js
@@ -43,17 +43,19 @@ export function applyConfigToArgv({
     }
 
     const decamelizedOptName = decamelize(option, '-');
-    let expectedType;
-    if (options[decamelizedOptName]) {
-      if (options[decamelizedOptName].type === undefined) {
-        throw new UsageError(
-          `Option: ${option} was defined without a type.`);
-      } else {
-        expectedType = options[decamelizedOptName].type ===
-          'count' ? 'number' : options[decamelizedOptName].type;
-      }
+
+    if (typeof options[decamelizedOptName] !== 'object') {
+      throw new UsageError(`The config file at ${configFileName} specified ` +
+        `an unknown option: "${option}"`);
+    }
+    if (options[decamelizedOptName].type === undefined) {
+      throw new WebExtError(
+        `Option: ${option} was defined without a type.`);
     }
 
+    const expectedType = options[decamelizedOptName].type ===
+      'count' ? 'number' : options[decamelizedOptName].type;
+
     if (configObject[option]) {
       const optionType = typeof(configObject[option]);
       if (optionType !== expectedType &&
@@ -85,11 +87,6 @@ export function applyConfigToArgv({
       continue;
     }
 
-    if (typeof options[decamelizedOptName] !== 'object') {
-      throw new UsageError(`The config file at ${configFileName} specified ` +
-        `an unknown option: "${option}"`);
-    }
-
     newArgv[option] = configObject[option];
   }
   return newArgv;
@@ -114,4 +111,4 @@ export function loadJSConfigFile(filePath: string): Object {
       'Did you set module.exports = {...}?');
   }
   return configObject;
-}
\ No newline at end of file
+}
```